server/ERNI.PBA.Server.DataAccess/Repository/UserRespository.cs

```diff
@@ -23,10 +23,23 @@ public Task<User> GetUser(int id, CancellationToken cancellationToken)
                 .FirstOrDefaultAsync(cancellationToken);
         }
 
-        public Task<User[]> GetUsers(CancellationToken cancellationToken)
+        /// <summary>
+        /// Gets the inferior users for the superior. 
+        /// If superior is admin, gets all users.
+        /// </summary>
+        public Task<User[]> GetInferiorUsers(int superiorId, CancellationToken cancellationToken)
         {
-            return _context.Users
-                .Include(u => u.Superior)
+            var user = GetUser(superiorId, cancellationToken).Result;
+
+            var users = _context.Users.Include(u => u.Superior);
+
+            if (user.IsAdmin)
```

Comment:
This is business logic, do not put it into repo. I propose to have two methods, one for returning all of the users, the other for returning just the subordinates.